pilot/pkg/xds/endpoint_builder.go

```diff
@@ -328,22 +322,21 @@ func (b *EndpointBuilder) buildLocalityLbEndpointsFromShards(
 			if ep.EnvoyEndpoint == nil {
 				ep.EnvoyEndpoint = buildEnvoyLbEndpoint(ep)
 			}
-			if features.EnableAutomTLSCheckPolicies {
-				tlsMode := ep.TLSMode
-				if b.mtlsChecker != nil && b.mtlsChecker.mtlsDisabledByPeerAuthentication(ep) {
-					tlsMode = ""
-				}
-				if nep, modified := util.MaybeApplyTLSModeLabel(ep.EnvoyEndpoint, tlsMode); modified {
-					ep.EnvoyEndpoint = nep
-				}
-			}
-			locLbEps.append(ep, ep.EnvoyEndpoint, ep.TunnelAbility)
-
 			// detect if mTLS is possible for this endpoint, used later during ep filtering
 			// this must be done while converting IstioEndpoints because we still have workload labels
 			if b.mtlsChecker != nil {
 				b.mtlsChecker.computeForEndpoint(ep)
+				if features.EnableAutomTLSCheckPolicies {
+					tlsMode := ep.TLSMode
+					if b.mtlsChecker != nil && b.mtlsChecker.isMtlsDisabled(ep.EnvoyEndpoint) {
```

Comment:
got it. the idea is that computeForEndpoint will contain a call of mtlsDisabledByPeerAuthentication, so by the line of 331, we don't have to re-calculate again, is that right?

that SG to me.

Reply:
thinking more. to avoid this re-invocation again, now i see all the calling of mTLSDisabledByPeerAuthn is used within computeEndpionts.

If we are putting the mTLSDisabledByPeerAuthn a local func variable within computeEndpiont, other code won't have access to the function right?

to me, it seems we only need look up via isMTLSdisabled.

Reply:
sgtm